### ISSUE TITLE
add @nuxt/image

### ISSUE THREAD

**Issue by yangwao** [kodadot/nft-gallery#7807]
Try tune and leverage all performance things we could use

- https://image.nuxt.com/
- https://image.nuxt.com/get-started/installation
- https://image.nuxt.com/providers/cloudflare

Ref
---
- https://github.com/kodadot/nft-gallery/issues/4716
- https://github.com/kodadot/nft-gallery/issues/4551

**Comment by roiLeo** [comment]
duplicate #4426

**Comment by pbkompasz** [comment]
:wave:

**Comment by kodabot** [comment]
ASSIGNED - @pbkompasz 🔒 LOCKED -> Thursday, November 2nd 2023, 09:56:31 UTC -> 24 hours

**Comment by pbkompasz** [comment]
@yangwao Some of the images link to `https://image.w.kodadot.xyz` or `https://image-beta.w.kodadot.xyz` which redirects to the Cloudfare provider. The problem is that the modifiers are lost this way.
To solve this I wrote a custom provider which fetches the images directly from `https://imagedelivery.net/...` with the modifiers, however when I make the requests I get this error:
`ERROR 9429: Flexible variants are not enabled for the account`
This is an example of an optimized image: `https://imagedelivery.net/jk5b6spi_m_-9qC4VTnjpg/bafybeiatgshrpwwv4cw3kdi5xzvxvbgwtf7hfelbz54vwjwfkdltvcra3e/public/width=200/height=100/fit=cover`

**Comment by yangwao** [comment]
> ERROR 9429: Flexible variants are not enabled for the account

hm I would enable it once CF is back

![image](https://github.com/kodadot/nft-gallery/assets/5887929/d2ef3469-2bbc-4228-8f71-eff98b222e29)




> ome of the images link to `https://image.w.kodadot.xyz` or `https://image-beta.w.kodadot.xyz` which redirects to the Cloudfare provider.

this is intentionally as we are having pre production environment and production.




> To solve this I wrote a custom provider

if it would respect when it's canary, beta or production I'm good

**Comment by yangwao** [comment]
Seems it was already on, maybe due CF outage it did not work? cc @pbkompasz 

- https://developers.cloudflare.com/images/cloudflare-images/transform/flexible-variants/

![image](https://github.com/kodadot/nft-gallery/assets/5887929/435e43af-98ec-49cf-8548-2223d58381df)

There is also this one fixed if you want to try

![image](https://github.com/kodadot/nft-gallery/assets/5887929/b03d89a6-8404-4fcb-8294-2694f7ed4f22)

**Comment by pbkompasz** [comment]
@yangwao It works now, probably it was due to the outage. I tested only a couple of optimization with making requests directly to `imagedelivery` and the explore page went down from ~35 MB to ~23 MB.
When I tried to make the requests to this [worker](https://github.com/kodadot/workers/blob/main/image/src/index.ts) it redirected to `cloudflare-ipfs.com` instead of `imagedelivery` which threw this error:
`failed to resolve /ipfs/QmYSgSKuP2oSpE4yBERxm7UGGWLwgpW9ywWwktx53hCAzy/width=240,height=240,fit=cover: no link named "width=240,height=240,fit=cover" under QmYSgSKuP2oSpE4yBERxm7UGGWLwgpW9ywWwktx53hCAzy` .
I will look into it tomorrow.
